crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -556,47 +511,124 @@ impl ScheduleGraph {
         &self.conflicting_systems
     }
 
-    fn add_systems<M>(&mut self, systems: impl IntoSystemConfigs<M>) {
-        let SystemConfigs { systems, chained } = systems.into_configs();
-        let mut system_iter = systems.into_iter();
-        if chained {
-            let Some(prev) = system_iter.next() else { return };
-            let mut prev_id = self.add_system_inner(prev).unwrap();
-            for next in system_iter {
-                let next_id = self.add_system_inner(next).unwrap();
-                self.dependency.graph.add_edge(prev_id, next_id, ());
-                prev_id = next_id;
+    /// Adds the systems to the graph. Returns a vector of all node ids contained the nested `SystemConfigs`
+    /// if `ancestor_chained` is true. Also returns true if "densely chained", meaning that all nested items
+    /// are linearly chained in the order they are defined
+    fn add_systems_inner(
+        &mut self,
+        configs: SystemConfigs,
+        ancestor_chained: bool,
+    ) -> (Vec<NodeId>, bool) {
```

Comment:
Can we make a simple named struct for this? This sort of tuple of primitives return type tends to spread messily and is hard to understand / easy to screw up.